Database-Features/explanation-of-the-security-manager-for-the-JDBC-driver-and-the-impact-of-disabling-it.md

```diff
@@ -0,0 +1,52 @@
+# Explanation of the Security Manager for the JDBC Driver and the Impact of Disabling It
+
+## Question
+
+You can disable the Security Manager by setting NOSECURITY=YES in the JDBC driver configuration, as shown in the following example:
+
+```text
+DRIVERNAME=MSSQLServer
+PREFIX=jdbc:sqlserver:
+FETCHSIZE=100000
+INSERTSIZE=-1
+NOSECURITY=YES
+```
+
+What is the security manager for the Exasol JDBC driver and what are the implications of disabling it in the JDBC driver configuration?
+
+## Answer
+
+### Background Information
+
+The Java Security Manager was introduced in Java 1.0.
+
+Its primary purpose was to provide a "sandbox" for Java applets, which were small Java applications embedded in web browsers. At the time, applets were a common way to deliver dynamic content on the web. The Java Security Manager's role was to protect the user's system from potentially malicious code downloaded from the internet.
+
+It enforced a security policy that defined what actions the untrusted applet code was allowed to perform. This included restricting access to local resources such as the file system, network connections, and system properties, preventing an applet from, for example, reading or deleting files on the user's computer without permission.
+
+The Java Security Manager has been deprecated for removal and is now effectively disabled in modern Java versions. It was deprecated in Java 17 (JEP 411) and permanently disabled in Java 24 (JEP 486). This means that even if you try to enable it via command-line options or by calling System.setSecurityManager(), it will either be ignored or an UnsupportedOperationException will be thrown.
+There were several key reasons for this change, as outlined in the JDK Enhancement Proposals (JEPs).
+
+Despite its original purpose, the Security Manager was rarely used to secure server-side applications. It was disabled by default, and developers found it notoriously difficult to configure correctly. A slight misconfiguration could either render the security useless or, more commonly, break the application entirely. Its all-or-nothing nature made it unappealing for the nuanced security needs of modern applications.
+
+### Changes to Java Security Manager Usage with Exasol Drivers
+
+Setting the NOSECURITY flag to YES disables Java Security Manager, allowing function calls without additional checks.
+
+The Java Security Manager provides detailed control over which functions are permitted to run in your environment. However, configuring these settings can be complex and time-consuming.  In the past, we tried to whitelist only the functions that JDBC drivers required. However, because the drivers frequently change and add new functions with each update, maintaining an accurate whitelist soon became impractical—especially since ExaLoader would then require a comprehensive list covering all functions for all drivers.
```

Comment:
While that's true in general, for the latest occurrences JDBC connection that IMPORT tried to establish just hung until one disables Security Manager. So we had no exception stating which class / method wasn't allowed so we had no particular item to whitelist.

So it's not only time consuming (many third party drivers) and in vain (because the component is deprecated by Java platform), but also simply not always possible. This aspect might show us in even better light, so maybe you can adapt the article accordingly?